lib/ffmpeg/transcoder.rb

```diff
@@ -36,16 +37,17 @@ def initialize(movie, output_file, options = EncodingOptions.new, transcoder_opt
       if requires_pre_encode
         @movie.paths.each do |path|
           # Make the interim path folder if it doesn't exist
-          dirname = "#{File.dirname(path)}/interim"
+
+          dirname = "#{TEMP_DIR}/interim/#{File.dirname(path)}/"
           unless File.directory?(dirname)
             FileUtils.mkdir_p(dirname)
           end
-
-          interim_path = "#{File.dirname(path)}/interim/#{File.basename(path, File.extname(path))}_#{SecureRandom.urlsafe_base64}.mp4"
+          interim_path = "#{dirname}#{File.basename(path, File.extname(path))}_#{SecureRandom.urlsafe_base64}.mp4"
           @movie.interim_paths << interim_path
         end
       else
         @movie.interim_paths << @movie.paths
+        @movie.interim_paths.flatten!
```

Comment:
Is this flatten needed? Seems like its already a 1d array of strings

Reply:
`@movie.paths` can either be `'/tmp/test.mp4` or an array `['/tmp/test.mp4']` for example, and it leads this to becoming a 2D array: `[['/tmp/test.mp4']]`

This then doesn't play nicely with some of the later things

Reply:
`requires_pre_encode?` would prevent this, due to this check [`@movie.paths.size > 1`](https://github.com/Vidyard/rlovelett-ffmpeg/pull/32/files/8aec9d26922ff919b976d1d0c8820b3b84713fe3#diff-315b5d89df7c1f1a332f88c975c4e36a1c978aa321fbb69c48bb1922ebc0bf6dR24)

Reply:
There are two other scenarios it can be called:
```
@movie.requires_pre_encode ||= @movie.paths.size > 1 || (@movie.has_dynamic_resolution && @transcoder_options[:permit_dynamic_resolution_pre_encode])
```
If `requires_pre_encode` was manually set to true by the callee, or `has_dynamic_resolution` is true 😛

Reply:
Sorry, Im not following here.
`@movie.requires_pre_encode` is only set by `requires_pre_encode` once its already been determined.
And having an array of movie paths would result in it being `true` preventing the line 49's ` @movie.interim_paths << @movie.paths` from ever happening.

Reply:
I'm not sure what you said makes any sense.
If Transcoder is initialized with an array of videos, even if it's only a single video, it may not pass the `requires_pre_encode` check and would therefore got to the `@movie.interim_paths << @movie.paths` code branch.

This can be seen in the existing rspecs for this suite which validates this works (and was how I caught this to add the flatten)

Reply:
Took me way too long to dig into this, but I see what you're saying now.
Interim paths is a 2d array when a single input is provided, BUT its never had any issues because we've never referenced interim paths UNLESS its >1 in size!
Now that we're using Interim paths when its size is 1, its causing issues, and flatten fixes it.